pkg/router/controller/route_secret_manager.go

```diff
@@ -0,0 +1,217 @@
+	}
+
+	// call next plugin
+	return p.plugin.HandleRoute(eventType, route)
+}
+
+// validateAndRegister validates the route's externalCertificate configuration and registers it with the secret manager.
+// It also updates the in-memory TLS certificate and key after reading from secret informer's cache.
+func (p *RouteSecretManager) validateAndRegister(route *routev1.Route) error {
+	fldPath := field.NewPath("spec").Child("tls").Child("externalCertificate")
+	// validate
+	if err := routeapihelpers.ValidateTLSExternalCertificate(route, fldPath, p.sarClient, p.secretsGetter).ToAggregate(); err != nil {
+		klog.Error(err, "skipping route due to invalid externalCertificate configuration", " route ", route.Name)
+		p.recorder.RecordRouteRejection(route, "ExternalCertificateValidationFailed", err.Error())
+		p.plugin.HandleRoute(watch.Deleted, route)
+		return fmt.Errorf("invalid route configuration for externalCertificate")
+	}
+
+	// register route with secretManager
+	secreth := p.generateSecretHandler(route)
+	if err := p.secretManager.RegisterRoute(context.TODO(), route.Namespace, route.Name, getReferencedSecret(route), secreth); err != nil {
+		klog.Error("failed to register route", err)
+		return err
+	}
+	// read referenced secret
+	secret, err := p.secretManager.GetSecret(context.TODO(), route.Namespace, route.Name)
+	if err != nil {
+		klog.Error("failed to read referenced secret", err)
+		return err
+	}
+	// update tls.Certificate and tls.Key
+	// NOTE: this will be in-memory change and won't update actual route resource
+	route.Spec.TLS.Certificate = string(secret.Data["tls.crt"])
+	route.Spec.TLS.Key = string(secret.Data["tls.key"])
+
+	return nil
+}
+
+// generateSecretHandler creates ResourceEventHandlerFuncs to handle Add, Update, and Delete events on secrets.
+// AddFunc: Invoked when a new secret is added. It logs the addition of the secret.
+// UpdateFunc: Invoked when an existing secret is updated. It performs validation of the route's external certificate configuration.
+// If the validation fails, it records the route rejection, and triggers the deletion of the route by calling the HandleRoute method with a watch.Deleted event.
+// If the validation succeeds, it updates the route's TLS certificate and key with the new secret data and calls the next plugin's HandleRoute method with a watch.Modified event, and then the next plugin's Commit() method.
+// DeleteFunc: Invoked when the secret is deleted. It unregisters the associated route, records the route rejection, and triggers the deletion of the route by calling the HandleRoute method with a watch.Deleted event.
+func (p *RouteSecretManager) generateSecretHandler(route *routev1.Route) cache.ResourceEventHandlerFuncs {
+	// secret handler
+	secreth := cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			secret := obj.(*kapi.Secret)
+			klog.Infof("secret %s added for %s/%s", secret.Name, route.Namespace, route.Name)
```

Comment:
```suggestion
			log.V(4).Info("secret added for route", "namespace", route.Namespace, "secret", secret.Name, "route", route.Name)
```

Reply:
Thanks, updated to `log` with correct log-level